File: apps/spring-boot-api/src/main/java/org/bootstrapbugz/api/user/service/ProfileService.java
package org.bootstrapbugz.api.user.service;

import org.bootstrapbugz.api.user.payload.request.ChangePasswordRequest;
import org.bootstrapbugz.api.user.payload.request.UpdateProfileRequest;
import org.bootstrapbugz.api.user.payload.response.UserResponse;

public interface ProfileService {
  UserResponse update(UpdateProfileRequest updateProfileRequest);

  void changePassword(ChangePasswordRequest changePasswordRequest);
}

File: frontend/src/app/user/services/profile.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { API_URL_PROFILE } from '../../shared/constants/paths';
import { User } from '../models/user.models';
import { ChangePasswordRequest, UpdateProfileRequest } from '../models/user.requests';

@Injectable({
  providedIn: 'root',
})
export class ProfileService {
  constructor(private http: HttpClient) {}

  update(updateProfileRequest: UpdateProfileRequest) {
    return this.http.put<User>(`${API_URL_PROFILE}/update`, updateProfileRequest);
  }

  changePassword(changePasswordRequest: ChangePasswordRequest) {
    return this.http.put<void>(`${API_URL_PROFILE}/change-password`, changePasswordRequest);
  }
}
